frontend/src/slices/auth/actions.ts

```diff
@@ -12,10 +13,24 @@ const signUp = createAsyncThunk<
   UserSignUpResponseDto,
   UserSignUpRequestDto,
   AsyncThunkConfig
->(`${sliceName}/sign-up`, (registerPayload, { extra }) => {
-  const { authApi } = extra;
+>(`${sliceName}/sign-up`, async (registerPayload, { extra }) => {
+  const { authApi, storage } = extra;
+  const response = await authApi.signUp(registerPayload);
+  await storage.set(StorageKey.TOKEN, response.token);
 
-  return authApi.signUp(registerPayload);
+  return response;
 });
 
-export { signUp };
+const signIn = createAsyncThunk<
+  UserSignUpResponseDto,
+  UserSignUpRequestDto,
+  AsyncThunkConfig
+>(`${sliceName}/sign-in`, async (loginPayload, { extra }) => {
+  const { authApi, storage } = extra;
+  const response = await authApi.signIn(loginPayload);
+  await storage.set(StorageKey.TOKEN, response.token);
+
+  return response;
```

Comment:
You've already set up token - there is no need to pass it deeper in code
```suggestion
  const { user, token } = await authApi.signIn(loginPayload);
  await storage.set(StorageKey.TOKEN, token);

  return user;
```